packages/react-devtools-shared/src/devtools/views/Profiler/CommitFlamegraph.js

```diff
@@ -115,9 +118,36 @@ function CommitFlamegraph({chartData, commitTree, height, width}: Props) {
     return null;
   }, [chartData, selectedFiberID, selectedChartNodeIndex]);
 
+  const highlightNativeElement = useCallback(
+    (id: number) => {
+      const element = store.getElementByID(id);
+      const rendererID = store.getRendererIDForElement(id);
+      if (element !== null && rendererID !== null) {
+        bridge.send('highlightNativeElement', {
+          displayName: element.displayName,
+          hideAfterTimeout: false,
+          id,
+          openNativeElementsPanel: false,
+          rendererID,
+          scrollIntoView: false,
+        });
+      }
+    },
+    [store, bridge],
+  );
+
+  // Highlight last hovered element.
+  const handleElementMouseEnter = useCallback(
+    id => {
+      highlightNativeElement(id);
+    },
+    [highlightNativeElement],
+  );
+
```

Comment:
This section is also repeated in `packages/react-devtools-shared/src/devtools/views/Profiler/CommitRanked.js` and also 
[packages/react-devtools-shared/src/devtools/views/Components/Tree.js](https://github.com/facebook/react/blob/e706721490e50d0bd6af2cd933dbf857fd8b61ed/packages/react-devtools-shared/src/devtools/views/Components/Tree.js).

Maybe better to create a separate hook file?

Reply:
Thanks for the comment, Yeah I absolutely agree,  I will try to use a hook for the highlighting functionality:)

Reply:
Done, now I extracted the highlighting logic to the hooks file, and created a hook called `useNativeElementHighlighter`, this hook exposes two methods: 
1. `highlightNativeElement`: method used to highlight an element using its id
2. `clearNativeElementHighlight`: method used to clear highlighting of elements